java/java.lsp.server/vscode/src/jdk/validation/project.ts

```diff
@@ -0,0 +1,171 @@
+import * as xml2js from 'xml2js';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as os from 'os';
+import { normalizeJavaVersion } from './javaUtil';
+
+const GET_PROJECT_INFO = 'nbls.project.info';
+const GRADLE_TARGET_COMPATIBILITY_REGEX = /targetCompatibility\s*=\s*(?:JavaVersion\.toVersion\(['"](\d+(\.\d+)?)['"]\)|['"](\d+(\.\d+)?)['"])/
```

Comment:
Nitpick: sadly the `JavaVersion.toVersion("string")` is a groovy script code, so there can be `\s*` between identifiers, literals and symbols. Unlikely to happen, though.